change thumbnail() to generateThumbnail() and DB cleanups

fvichot opened the pull request:
Implement the changes discussed in #225, as well as some DB cleanups based on #227 , notably I removed all notion of the "image" table in the DB.

rdebroiz commented:
When you saved an image in the persistent DB the thumbnail is not saved, I suspect that you are trying to write the QImage in a non existing folder.

fvichot commented:
**Make sure to backup your "db" file before you test this, as it will modify the schema of the series table, and drop the image table.**

So this as far as I got:
 - I've removed all traces of the "image" table
 - I've migrated the isIndexed field from image to series
 - the path field in series table is now used for the image(s) path(s). If more than one path, they are separated with ';'. This way we should retain the ability to index multi-files data items, like DICOM (not an issue in case of import as we convert it to a single file, a MHA in the case of DICOM).

If someone continues on this, one of the tricky bits is not breaking the whole DB, have a look at how I've implemented ```medDatabaseController::updateFromNoVersionToVersion1()```. Also, you'll notice I've started versioning the DB schema using ```PRAGMA user_version``` which seems like a good idea.
 This PR should also be merged with PR #207 at some point too, and as such it would provide a nicer base to refactor the importer during the upcoming data refactoring.
